### DISCUSSION TITLE
Discreet LED control

### DISCUSSION THREAD

**Discussion by trlafleur** [arendst/Tasmota#15054] [Q&A] [answered]
If I have a discreet LED (On-Off only) connected to my ESP32, and I want to control it from a new sensor driver code I'm developing, what is the proper way to control it from my code?

thanks...

**Comment by barbudor** [comment, marked as answer]
If you want to have an output directly attached to the driver, you'll need to define a  GPIO for it like
D_GPIO_TOM_LED
In the init function of the driver, you test if gpio as been assigned and if yes you configure as an output. Then use digitalWrite() to control.

Not on my PC now but if you want more detailed code, and can do it in  a few hours

**Reply by trlafleur** [reply]
I can't assign a LED that is defined already in Tasmota and control it from my driver??

**Reply by barbudor** [reply]
Led assigned in Tasmota already have  behavior
Like power or link
I would unadvise to mess with it from a driver
If this is a feature of your driver like a serial Com would blink a led each time it receive a frame, then IMHO that should be a GPIO for your driver

Alternatively if your driver push an event that can be captured in a Rule, you could do that with LedPower in a Rule.

Just to be sure, native C driver or Berry?

**Comment by trlafleur** [comment]
in C ....   I'll just add it as an option to my driver... Does serial I/O
have an example of this ??

~~ _/) ~~~~ _/) ~~~~ _/) ~~~~ _/) ~~

Tom Lafleur

**Reply by barbudor** [reply]
No, not in serial.
There's one of the VL5XLX driver that is managing output pins called XSHUT
Might give you some idea
